public/app/features/templating/variable_srv.ts

```diff
@@ -303,6 +311,42 @@ export class VariableSrv {
     this.$location.search(params);
   }
 
+  storeUpdated() {
+    const state = store.getState();
+    let path = this.originalUrl;
+    if (path.indexOf('?') >= 0) {
+      path = path.slice(0, this.originalUrl.indexOf('?'));
+    }
+    if (state.location.path === path) {
+      if (state.location.url === this.originalUrl) {
+        return this.$q.when({});
+      }
+
+      this.originalUrl = this.$location.url();
+      const queryParams = state.location.query; //this.$location.search();
+      if (!Object.keys(queryParams).some(k => k.indexOf('var-') === 0)) {
+        return this.$q.when({});
+      }
+      console.log('VAR Changed', queryParams);
+      return this.$q
+        .all(
+          this.variables.map(variable => {
+            return this.processVariable(variable, queryParams);
+          })
+        )
+        .then(() => {
+          this.templateSrv.updateIndex();
+          this.dashboard.templateVariableValueUpdated();
+          this.dashboard.startRefresh();
```

Comment:
We can't do this on every location update. Not sure why index needs updating, and we cant issue a full dashboard refresh on every location change without identifying if a variable value was actually changed.